exercises/practice/queen-attack/queen_attack_test.go

```diff
@@ -1,51 +1,62 @@
 package queenattack
 
-import "testing"
+import (
+	"testing"
+)
 
 // Arguments to CanQueenAttack are in algebraic notation.
 // See http://en.wikipedia.org/wiki/Algebraic_notation_(chess)
 
-var tests = []struct {
-	w, b   string
-	attack bool
-	ok     bool
-}{
-	{"b4", "b4", false, false}, // same square
-	{"a8", "b9", false, false}, // off board
-	{"a0", "b1", false, false},
-	{"g3", "i5", false, false},
-	{"here", "there", false, false}, // invalid
-	{"", "", false, false},
-
-	{"b3", "d7", false, true}, // no attack
-	{"a1", "f8", false, true},
-	{"b4", "b7", true, true}, // same file
-	{"e4", "b4", true, true}, // same rank
-	{"a1", "f6", true, true}, // common diagonals
-	{"a6", "b7", true, true},
-	{"d1", "f3", true, true},
-	{"f1", "a6", true, true},
-	{"a1", "h8", true, true},
-	{"a8", "h1", true, true},
+type testCase struct {
+	description string
+	pos1, pos2  string
+	expected    bool
 }
 
-func TestCanQueenAttack(t *testing.T) {
-	for _, test := range tests {
-		switch attack, err := CanQueenAttack(test.w, test.b); {
-		case err != nil:
-			var _ error = err
-			if test.ok {
-				t.Fatalf("CanQueenAttack(%s, %s) returned error %q.  "+
-					"Error not expected.",
-					test.w, test.b, err)
+var validTestCases = []testCase{
+	{description: "no attack", pos1: "b3", pos2: "d7", expected: false},
+	{description: "no attack", pos1: "a1", pos2: "f8", expected: false},
+	{description: "same file", pos1: "b4", pos2: "b7", expected: true},
+	{description: "same rank", pos1: "e4", pos2: "b4", expected: true},
+	{description: "common diagonals", pos1: "a1", pos2: "f6", expected: true},
+	{description: "common diagonals", pos1: "a6", pos2: "b7", expected: true},
+	{description: "common diagonals", pos1: "d1", pos2: "f3", expected: true},
+	{description: "common diagonals", pos1: "f1", pos2: "a6", expected: true},
+	{description: "common diagonals", pos1: "a1", pos2: "h8", expected: true},
+	{description: "common diagonals", pos1: "a8", pos2: "h1", expected: true},
+}
+
+func TestCanQueenAttackValid(t *testing.T) {
+	for _, tc := range validTestCases {
+		t.Run(tc.description, func(t *testing.T) {
```

Comment:
I would recommend to add the position to the description so it is unique, otherwise Go does this which might be confusing for students:
```
    --- PASS: TestCanQueenAttackValid/common_diagonals (0.00s)
    --- PASS: TestCanQueenAttackValid/common_diagonals#01 (0.00s)
    --- PASS: TestCanQueenAttackValid/common_diagonals#02 (0.00s)
    --- PASS: TestCanQueenAttackValid/common_diagonals#03 (0.00s)
    --- PASS: TestCanQueenAttackValid/common_diagonals#04 (0.00s)
    --- PASS: TestCanQueenAttackValid/common_diagonals#05 (0.00s)
```

If our test cases match the problem-spec test cases, we could use the names from there as well:
https://github.com/exercism/problem-specifications/blob/main/exercises/queen-attack/canonical-data.json

Reply:
Added the position to the name